pwiz_tools/Skyline/Model/AbstractDdaSearchEngine.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using pwiz.Skyline.Model.DocSettings;
+
+namespace pwiz.Skyline.Model
+{
+    public abstract class AbstractDdaSearchEngine
+    {
+        public abstract string[] FragmentIons { get; }
+        public abstract string EngineName { get; }
+        public abstract Bitmap SearchEngineLogo { get; }
+        protected string[] SpectrumFileNames { get; set; }
+        protected string[] FastaFileNames { get; set; }
+
+        public abstract void SetMS1Tolerance(double mass, string unit);
```

Comment:
It's unfortunate Skyline doesn't have an MzTolerance class right now. There's MassTolerance in FastOverlapDemultiplexer but it's ppm-only. If we had the class, I'd say this should be a get/set property instead of SetXX() function. Same with the others below. MaxMissedCleavages could be its own property, and there will be search-engine-specific settings to add, somehow. Perhaps via a dictionary here? I guess we need to decide whether each search engine will get its own UI control or whether they will all share a single generic control.

Reply:
true, this is an important decision. MS Amanda has some additional settings, that are left at default right now, but I added a button for additional settings that could open kind of a table view with the name of the parameter and its value to be changed and those could be read from the dictionary?!